parts/eventsets/master_g_modern.lua

```diff
@@ -0,0 +1,406 @@
+-- local regretDelay=-1
+-- local int_grade=0
+-- local grade_points=0
+local _igb={0,1,2,3,3,4,4,5,5,5,6,6,6,7,7,7,8,8,8,9,9,9,10,10,11,11,12,12,12,13,13,14,14,15,15,16,16,17,17,18,18,19,19,20,20,21,21,22,22,23,23,24,24,25,25,26}
+local function getInternalGradeBoosts(internal_grade)
+    return _igb[MATH.clamp(internal_grade+1,1,#_igb)]
+end
+-- local spd_lvl=0
+-- local cools=0
+-- local regrets=0
+-- local prevSectTime=0
+-- local isInRoll=false
+-- local rollGrades=0
+local awesome_time=    {2700,2700,2520,2280,2280,1980,1980,1620,1620,0}
+local cool_time=   {3120,3120,2940,2700,2700,2520,2520,2280,2280,0}
+local regret_time= {5400,4500,4500,4080,3600,3600,3000,3000,3000,3000}
+local fault_time=    {7200,5400,5400,4500,4080,4080,3600,3600,3600,3600}
+local function getGrav(l)
+    return
+        l<30  and 64   or
+        l<40  and 43   or
+        l<50  and 32   or
+        l<60  and 26   or
+        l<70  and 21   or
+        l<80  and 16   or
+        l<90  and 8    or
+        l<120 and 6    or
+        l<160 and 4    or
+        l<200 and 3    or
+        l<240 and 2    or
+        l<260 and 64   or
+        l<270 and 8    or
+        l<290 and 4    or
+        l<320 and 3    or
+        l<370 and 2    or
+        l<430 and 1    or
+        l<510 and 0.5  or
+        l<600 and 0.25 or
+        l<700 and 0.1  or
```

Comment:
Welp I have tested again, I got issue at lv430
If I were you then I would make the speed curve go down smoother

This is the reference, I'm just using some random numbers, not sure if it is a good curve
```lua
        l<260 and 64   or
        l<268 and 32   or
        l<275 and 16   or
        l<300 and 4    or
        l<320 and 3    or
        l<370 and 2    or
        l<430 and 1    or
        l<480 and 0.75 or
        l<530 and 0.5  or
        l<600 and 0.25 or
        l<700 and 0.1  or
```

Reply:
Change speed sets to `
        l<30  and 64   or
        l<40  and 43   or
        l<50  and 32   or
        l<60  and 26   or
        l<70  and 21   or
        l<80  and 16   or
        l<90  and 8    or
        l<120 and 6    or
        l<160 and 4    or
        l<200 and 3    or
        l<240 and 2    or
        l<260 and 64   or
        l<268 and 32   or
        l<275 and 16   or
        l<280 and 8    or
        l<290 and 4    or
        l<320 and 3    or
        l<370 and 2    or
        l<430 and 1    or
        l<470 and 0.75 or
        l<510 and 0.5  or
        l<600 and 0.25 or
        l<700 and 0.1  or`